scripts/utils/logger.js

```diff
@@ -0,0 +1,48 @@
+const chalk = require('chalk');
```

Comment:
I had a few ideas that were a little hard to explain. I've made a gist with what I had in mind. 

I think the formatting of the logger can expand to allow all of the normal chalk options. Example: `chalk.bold.underline.red('Message')` 

The main changes are combining getMessage, setContext, getDetails into a single API.

```js
/**
 * Formats a message.
 * @param {string} message - The message to format.
 * @param {Object} options - The message options.
 * @param {Object} decorations - The decoration options (bold, underline, color, etc...).
 * @returns - A formatted message.
 */
static format(message, options, decorations) {
  // Code
}
```

And adding a decorate function:

```js
/**
 * Decorates a message with the provided options.
 * @param {string} message - The message to decorate.
 * @param {Object} decorations - The decoration options.
 */
static decorate(message, decorations) {
  // Code
}
```
https://gist.github.com/StephenEsser/bc6b7ff1d2f9b890797030528048410e

Reply:
This is missing the emphasis function, but let me know what your thoughts are.

Reply:
For anyone curious,  conversation on Logger details are on the gist Stephen provided & on here:  https://gist.github.com/emilyrohrbough/665d93aff112323110fda2fbc255d019